reconcile/external_resources/secrets_sync.py

```diff
@@ -450,7 +450,6 @@ def _read_secret(self, spec: ExternalResourceSpec) -> None:
             logging.debug("Reading Secret %s", secret_path)
             data = self.secrets_reader.read_all({"path": secret_path})
             spec.metadata[SECRET_UPDATED_AT] = data[SECRET_UPDATED_AT]
-            del data[SECRET_UPDATED_AT]
-            spec.secret = data
+            spec.secret = {k: v for k, v in data.items() if k != SECRET_UPDATED_AT}
```

Comment:
please add a test case, and to avoid mutation and just pick one key value out of dict, we can just copy it

```python
            data = self.secrets_reader.read_all({"path": secret_path}).copy()
            spec.metadata[SECRET_UPDATED_AT] = data.pop(SECRET_UPDATED_AT)
            spec.secret = data

```